extra-tests/classes/RollupStateTests.cls

```diff
@@ -0,0 +1,371 @@
+@IsTest
+private class RollupStateTests {
+  @IsTest
+  static void commitsAndLoadsStateProperly() {
+    RollupState state = new RollupState();
+    String stubAccountId = RollupTestUtils.createId(Account.SObjectType);
+    RollupState.GenericInfo info = (RollupState.GenericInfo) state.getState(
+      stubAccountId,
+      new Rollup__mdt(RollupOperation__c = 'SUM'),
+      RollupState.GenericInfo.class
+    );
+    info.value = 5;
+
+    RollupState.AverageInfo averageInfo = (RollupState.AverageInfo) state.getState(
+      stubAccountId,
+      new Rollup__mdt(RollupOperation__c = 'CONCAT'),
+      RollupState.AverageInfo.class
+    );
+    averageInfo.increment(10);
+    RollupState.SObjectInfo sObjectInfo = (RollupState.SObjectInfo) state.getState(
+      stubAccountId,
+      new Rollup__mdt(RollupOperation__c = 'FIRST'),
+      RollupState.SObjectInfo.class
+    );
+    sObjectInfo.setItem(new Account(AnnualRevenue = 1000));
+    String secondStubId = RollupTestUtils.createId(Contact.SObjectType);
+    RollupState.MostInfo mostInfo = (RollupState.MostInfo) state.getState(
+      secondStubId,
+      new Rollup__mdt(RollupOperation__c = 'MOST'),
+      RollupState.MostInfo.class
+    );
+    mostInfo.setValues(5, 'some string');
+    // populate a null state value to be sure if the last iteration is "empty" the body is still properly committed
+    state.getState(secondStubId, new Rollup__mdt(RollupOperation__c = 'LAST'), RollupState.GenericInfo.class);
+
+    Id stubJobId = RollupTestUtils.createId(AsyncApexJob.SObjectType);
+
+    Set<String> relatedRecordKeys = new Set<String>{ '%' + stubAccountId + '%', '%' + secondStubId + '%' };
+    Test.startTest();
+    state.commitState(stubJobId);
+    state.loadState(stubJobId, new Set<String>{ stubAccountId, secondStubId });
+    info = (RollupState.GenericInfo) state.getState(stubAccountId, new Rollup__mdt(RollupOperation__c = 'SUM'), RollupState.GenericInfo.class);
+    info.value = 6;
+    state.commitState(stubJobId);
+    Test.stopTest();
```

Comment:
Using `Test.startTest()` and Test.stopTest()` don't seem to be needed? I commented out those lines & the test still passes.

Reply:
Good call, I can't remember if originally here I was planning to chain this all into the cleanup method, but I can remove those lines now!